app/src/main/java/com/github/geohunt/app/ui/screens/login/LoginViewModel.kt

```diff
@@ -103,16 +103,31 @@ class LoginViewModel(
     }
 
     companion object {
+        private var injectedAuthRepo: AuthRepositoryInterface? = null
+        private var injectedUserRepo: UserRepositoryInterface? = null
+
         val Factory: ViewModelProvider.Factory = viewModelFactory {
             initializer {
                 val application = this[ViewModelProvider.AndroidViewModelFactory.APPLICATION_KEY] as Application
                 val container = AppContainer.getInstance(application)
 
                 LoginViewModel(
-                    container.auth,
-                    container.user
+                    injectedAuthRepo ?: container.auth,
+                    injectedUserRepo ?: container.user
                 )
             }
         }
+
+        /**
+         * Used to inject custom repositories for testing
+         */
+        fun injectRepos(authRepository: AuthRepositoryInterface?, userRepository: UserRepositoryInterface?) {
+            injectedAuthRepo = authRepository
+            injectedUserRepo = userRepository
+        }
+
+        fun uninjectRepos() {
+            injectRepos(null, null)
+        }
```

Comment:
I think that there is no need to inject the repositories this way, as you could declare the factory as `open` and mock it to return a test `LoginViewModel`.

Reply:
This would be uselessly heavy in my opinion.